script2.js

```diff
@@ -1,18 +1,36 @@
-/********************
-* Objects and Properties
-*/
-
 var john = {
-    firstname: 'John',
-    birthYear: 1994,
-    family: ['Jane', 'Mark'],
-    isMarried: false,
-    calcAge: function(){
-        this.age = 2018 - this.birthYear; //this
+    fullName : 'John Burrows',
+    mass : 80,
+    height : 1.8,
+    calcBMI: function(){
```

Comment:
Exercise: calcBMI is defined now twice. How to make it work so that it's defined only once, but "this" works?

Reply:
Hint: `bind`

Reply:
Good excercise thanks.
It's much cleaner also that I can now use mark.BMI and john.BMI directly

Reply:
yeah, it's better, but not sulanto-perfect, because now it introduces an unnecessary proprty BMI (BMI is derivate of two other variables, so it would be better to calculate that on demand). 

In this exercise it doesn't make a difference, but in real life the similar things are causing bugs, imagine you update the height but then forget to re-calculate the BMI property.

catch my drift?

Reply:
Makes sense thanks. On the other side of the coin is trying to keep the code easy to read. I've just been told that it is usually better to create a new variable instead of repeating the same code. So in this case it would be clearer to do calculations with 'BMI' rather than 'this.mass / (this.height * this.height)'

Reply:
you would be using "john.bmi()"

Reply:
having `'this.mass / (this.height * this.height)'` everywhere would be plain awful

Reply:
ah ok. so calling john.bmi() would recalculate each time using current data. got it.